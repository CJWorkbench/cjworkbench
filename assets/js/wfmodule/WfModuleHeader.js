import React from 'react';
import PropTypes from 'prop-types';
import StatusBar from './StatusBar';

export default class WfModuleHeader extends React.Component {
  constructor(props) {
    super(props);
    this.setModuleRef = this.setModuleRef.bind(this);
  }

  componentDidMount() {
    if (this.props.isSelected) {
      this.props.focusModule(this.moduleRef);
    }
  }

  setModuleRef(ref) {
    this.moduleRef = ref;
  }

  shouldComponentUpdate(nextProps) {
    // There's almost certainly a better way to do this. This gets the
    // job done but I'm sure there's a standard practice around this.
    return nextProps.moduleName !== this.props.moduleName &&
      nextProps.isSelected !== this.props.isSelected;
  }

  render() {
    return (
      <div className='wf-module--placeholder mx-auto' ref={this.setModuleRef}>
        <div className='module-content'>
            <div className='module-card-header'>
              <div className='d-flex justify-content-start align-items-center'>
                <div className={'icon-' + this.props.moduleIcon + ' WFmodule-icon t-vl-gray mr-2'} />
                <div className='t-vl-gray WFmodule-name'>{this.props.moduleName}</div>
              </div>
            </div>
        </div>
      </div>
    );
  }
}

WfModuleHeader.PropTypes = {
  isSelected: PropTypes.bool,
  moduleIcon: PropTypes.string,
  moduleName: PropTypes.string,
  focusModule: PropTypes.func
};
