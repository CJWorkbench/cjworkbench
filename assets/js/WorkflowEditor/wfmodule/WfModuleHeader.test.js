/* globals describe, it, expect */
import React from 'react'
import WfModuleHeader from './WfModuleHeader'
import { mount } from 'enzyme'

describe('WfModuleHeader', () => {
  it('Mounts selected', () => {
    const wrapper = mount(
      <WfModuleHeader
        status='busy'
        isSelected
        moduleName='Some module name'
        moduleIcon='some-icon'
      />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.WFmodule-name').text()).toEqual('Some module name')
    expect(wrapper.find('.WFmodule-icon').props().className).toEqual('icon-some-icon WFmodule-icon t-vl-gray mr-2')
  })

  it('Mounts unselected', () => {
    const wrapper = mount(
      <WfModuleHeader
        status='busy'
        isSelected={false}
        moduleName='Some module name'
        moduleIcon='some-icon'
      />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.WFmodule-name').text()).toEqual('Some module name')
    expect(wrapper.find('.WFmodule-icon').props().className).toEqual('icon-some-icon WFmodule-icon t-vl-gray mr-2')
  })
})
