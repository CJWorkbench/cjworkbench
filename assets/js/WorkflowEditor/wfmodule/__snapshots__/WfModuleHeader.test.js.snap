// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WfModuleHeader Mounts selected 1`] = `
<WfModuleHeader
  isSelected={true}
  moduleIcon="some-icon"
  moduleName="Some module name"
  status="busy"
>
  <div
    className="wf-module--placeholder mx-auto"
  >
    <div
      className="module-content"
    >
      <div
        className="module-card-header"
      >
        <div
          className="d-flex justify-content-start align-items-center"
        >
          <div
            className="icon-some-icon WFmodule-icon t-vl-gray mr-2"
          />
          <div
            className="t-vl-gray WFmodule-name"
          >
            Some module name
          </div>
        </div>
      </div>
    </div>
  </div>
</WfModuleHeader>
`;

exports[`WfModuleHeader Mounts unselected 1`] = `
<WfModuleHeader
  isSelected={false}
  moduleIcon="some-icon"
  moduleName="Some module name"
  status="busy"
>
  <div
    className="wf-module--placeholder mx-auto"
  >
    <div
      className="module-content"
    >
      <div
        className="module-card-header"
      >
        <div
          className="d-flex justify-content-start align-items-center"
        >
          <div
            className="icon-some-icon WFmodule-icon t-vl-gray mr-2"
          />
          <div
            className="t-vl-gray WFmodule-name"
          >
            Some module name
          </div>
        </div>
      </div>
    </div>
  </div>
</WfModuleHeader>
`;
